libs/datagrid/src/lib/ag-grid/ag-grid-lazy-themed.tsx

```diff
@@ -1,30 +1,56 @@
+import React, { useCallback, useRef } from 'react';
+import type { ReactNode } from 'react';
 import type { AgGridReactProps, AgReactUiProps } from 'ag-grid-react';
+import type { ColumnResizedEvent } from 'ag-grid-community';
 import { AgGridReact } from 'ag-grid-react';
 import { useThemeSwitcher } from '@vegaprotocol/react-helpers';
+import { useColumnSizes } from './use-column-sizes';
 import classNames from 'classnames';
 
 export const AgGridThemed = ({
   style,
   gridRef,
+  id,
+  children,
   ...props
 }: (AgGridReactProps | AgReactUiProps) & {
   style?: React.CSSProperties;
   gridRef?: React.ForwardedRef<AgGridReact>;
+  id?: string;
+  children?: ReactNode[];
 }) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const [setValues, reshapeChildren, reshapeProps] = useColumnSizes({
```

Comment:
I don't think we should be enforcing every grid to store columns. It should be opt in via prop, or the hook should be used in each parent component of the table

Reply:
It is opt in, it's enableable via `id` property which is also the key in the store

Reply:
~~`id`~~ -> `storeKey`

Reply:
done